plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java

```diff
@@ -1541,6 +1541,23 @@ public void testPartitionProjectionIgnore()
                 "VALUES ('POLAND_1', '2022-02-01 12'), ('POLAND_2', '2022-02-01 12')");
     }
 
+    @Test
+    public void testSetColumnType()
+    {
+        String tableName = "test_set_column_type_" + randomNameSuffix();
+        String trinoTableName = getFullyQualifiedTestTableName(tableName);
+        String hiveTableName = getHiveTestTableName(tableName);
+
+        assertUpdate("CREATE TABLE " + trinoTableName + " AS SELECT CAST(123 AS integer) AS col", 1);
+
+        assertUpdate("ALTER TABLE " + trinoTableName + " ALTER COLUMN col SET DATA TYPE bigint");
+
+        assertQuery("SELECT * FROM " + trinoTableName, "VALUES 123");
+        assertThat(onHive("SELECT * FROM " + hiveTableName)).contains("123"); // Use contains method because the result includes table and column names
```

Comment:
Things like that should be exercised by a normal product test.
Then we work with normal JDBC result sets, and String#contains needed.